controllers/gitrepository_controller.go

```diff
@@ -714,6 +719,40 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 		checkoutOpts.SemVer = ref.SemVer
 	}
 
+	// managed GIT transport only affects the libgit2 implementation
+	if enabled, ok := r.features[features.GitManagedTransport]; ok && enabled &&
+		obj.Spec.GitImplementation == sourcev1.LibGit2Implementation {
+		// We return a stalling error if managed transport isn't registered and the related
+		// feature is enabled, since the controller can't recover from this.
+		if !r.ManagedTransportRegistered {
+			e := &serror.Stalling{
+				Err:    errors.New("invalid state: GitManagedTransport is enabled but managed transport isn't registered"),
+				Reason: sourcev1.ControllerMisbehaviorReason,
+			}
+			conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
+			return nil, e
+		}
```

Comment:
Seeing the complexity this new approach (removing `Enabled()` from libgit2/managed/ package and storing the result of initialization in `ManagedTransportRegistered`) adds, and considering that we'll not need all these anymore soon, maybe we don't need to change any of it.
In this implementation, what was short and simple before using `Enabled()` which indicated if managed transport is requested and is successfully initialized, is now divided into two separate variables and we now need to add more checks to ensure that the reconciler doesn't enter into an undefined state.
Also considering that in the current implementation, if managed transport initialization fails, Enabled() wouldn't be true, maybe we should keep Enabled() and improve it a little. https://github.com/fluxcd/source-controller/blob/f6a389ce5743a26649ae9d074a9b45a4ded594e3/pkg/git/libgit2/managed/init.go#L78-L79 will set enabled to true even if ssh managed transport initialization fails. We can make sure that enabled is true only when both http and ssh managed transports are initialized successfully.